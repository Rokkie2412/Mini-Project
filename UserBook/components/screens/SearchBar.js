import React from 'react'
import {StyleSheet,TextInput,View} from 'react-native'
import Search from 'react-native-vector-icons/Ionicons'
import SearchBar from "react-native-dynamic-search-bar";

const SearchBarComponent = ({search,setSearch,searchSubmit,clear}) => {
    return(
        <View style={styles.searchContainer}>
            {/* <Search style={styles.magnifyingGlass} name='search' size={28}/>
            <TextInput
                autoCapitalize='none'
                autoCorrect={false}
                placeholder='Search contact'   
                value={search}
                onChangeText={setSearch}
                style={{flex:1,fontSize:17}}
                onEndEditing={searchSubmit}
            /> */}
            <SearchBar
                style={{backgroundColor:'#F0EEEE',height:55,justifyContent:'center'}}
                height={50}
                fontSize={18}
                placeholder="Search Contact"
                placeholderTextColor='grey'
                backgroundColor='#F0EEEE'
                onClearPress={clear}
                value={search}
                onChangeText={setSearch}
                onEndEditing={searchSubmit}
            />
        </View>
    )
}

const styles = StyleSheet.create({
    searchContainer:{
        justifyContent:'center',
        height:50,
        borderRadius:10,
        marginVertical:10,
        // marginHorizontal:2,
        flexDirection:'row'
    },
    magnifyingGlass:{
        alignSelf:'center',
        color:'black',
        marginHorizontal:10,
    }
})

export default SearchBarComponent