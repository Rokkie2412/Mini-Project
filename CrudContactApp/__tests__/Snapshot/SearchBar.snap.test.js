import {create,act} from 'react-test-renderer';
import React from 'react';
import SearchBar from "../../src/components/Searchbar";

const func = jest.fn()
const tree = create(<SearchBar setSearch={func}/>)


jest.mock('react-native-vector-icons/Ionicons', () => 'Mocked Ionicons');

describe('Testing for SearchBar', () => {
  describe('Snapshot for SearchBar', () => {
    it('Should render correctly',()=>{
        expect(tree).toMatchSnapshot()
    })
  })
  describe('Testing Textinput for SearchBar', () => {
    it('Should correct onChangeText',()=>{
        const Search = tree.root.findByProps({testID:'Search'}).props
        act(()=>Search.onChangeText())
    })
  })
  
})
