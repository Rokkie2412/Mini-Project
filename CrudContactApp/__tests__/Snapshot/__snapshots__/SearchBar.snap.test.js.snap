// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for SearchBar Snapshot for SearchBar Should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393E46",
      "borderRadius": 9,
      "flexDirection": "row",
      "height": 50,
      "marginBottom": 10,
      "marginHorizontal": 15,
      "marginTop": 10,
    }
  }
>
  <Mocked Ionicons
    name="search"
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 30,
        "marginHorizontal": 15,
      }
    }
  />
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[Function]}
    placeholder="Search"
    placeholderTextColor="#Ffff"
    style={
      Object {
        "flex": 1,
        "fontSize": 18,
      }
    }
    testID="Search"
  />
</View>
`;
